### PR TITLE
Fixing null pointer on email not found

### PR DESCRIPTION
**Pull request by TonyBrobston**
Trying to fix a NullPointerException. I've returned a 404; I think in most cases spring's validation that I'm using for other things is returning a generic 400 for all failures (or so I think). Not sure if we want to just go with a 400 or a 404...?

Also spring's validation returns nice error messages that give you options, I'm not sure how to code things to use spring in this case. I'd prefer the output be consistent, I'm tempted to roll it myself and follow spring's syntax like this:
```
{
  "status": 404,
  "error": "Not Found",
  "errors": [
    {
      "codes": [
        "NotFound.user.email"
      ],
      "arguments": [
        {
          "codes": [
            "user.email",
            "email"
          ],
          "arguments": null,
          "defaultMessage": "email",
          "code": "email"
        }
      ],
      "defaultMessage": "email not found",
      "objectName": "project",
      "field": "name",
      "rejectedValue": "rejected value here",
      "code": "Not Found"
    }
  ],
  "message": "Validation failed for object='user'. Error count: 1",
  "path": "/login/SendCode"
}
```

Thoughts? Is a 404 enough? Should I just give a generic `{error: "Email not found"}`?

### CODE REVIEW

**Comment by TonyBrobston** [comment]
Not that it's a particularly good idea, I could call a rest endpoint from a rest endpoint to then use `@Validated`.

Instead of calling:
```
@PostMapping("/login/sendCode")
public ResponseEntity login(@RequestBody String emailAddress) {
    User user = userRepository.findByEmail(emailAddress);
    if (user == null) {
        return new ResponseEntity(HttpStatus.NOT_FOUND);
    }
    loginService.login(user);
    return new ResponseEntity(HttpStatus.CREATED);
}
```

Instead of:
```
if (user == null) {
    return new ResponseEntity(HttpStatus.NOT_FOUND);
}
```
I could make an api call to

**Comment by TonyBrobston** [comment]
I'm thinking pretty strongly about making the above json a reality through creating a handful of objects.

Something like `new ValidationError().setStatus(404).setObjectType(User).setField("email").setValue("test@aol.com").setPath("/login/sendCode").toJson();`

Might do this in a future PR, might add it to this one; we'll see.

File: src/main/java/org/dsmhack/controller/LoginController.java [hunk 1]
```diff
@@ -34,6 +34,9 @@
     @PostMapping("/login/sendCode")
     public ResponseEntity login(@RequestBody String emailAddress) {
         User user = userRepository.findByEmail(emailAddress);
+        if (user == null) {
```
Comment (AndreasChristianson) [inline src/main/java/org/dsmhack/controller/LoginController.java]: Optionals might shine in situations like this. More personal preference than anything else.

File: src/test/java/org/dsmhack/controller/LoginControllerTest.java [hunk 2]
```diff
@@ -36,22 +37,37 @@
...
+
+        MvcResult mvcResult = mockMvc.perform(
+                post("/login/sendCode")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("test@aol.com")
+        ).andExpect(
+                status().isNotFound()
+        ).andReturn();
+    }
 }
```
Comment (AndreasChristianson) [inline src/test/java/org/dsmhack/controller/LoginControllerTest.java]: omg missing newline!
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: src/main/java/org/dsmhack/controller/LoginController.java (hunk 1 above)
Comment (TonyBrobston) [reply src/main/java/org/dsmhack/controller/LoginController.java]: I'm not familiar with Optionals, I'll see if I can look it up a little later.

File: src/test/java/org/dsmhack/controller/LoginControllerTest.java (hunk 2 above)
Comment (TonyBrobston) [reply src/test/java/org/dsmhack/controller/LoginControllerTest.java]: Lol, I'll add one.

**Pull request merged**